specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-07-01-preview/authorization-DenyAssignmentGetCalls.json

```diff
@@ -0,0 +1,494 @@
+            "schema": {
+              "$ref": "#/definitions/DenyAssignment"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "GetConfigurations": {
+            "$ref": "./examples/GetDenyAssignmentByNameId.json"
+          }
+        }
+      }
+    },
+    "/{denyAssignmentId}": {
+      "get": {
+        "tags": [
+          "DenyAssignments"
+        ],
+        "operationId": "DenyAssignments_GetById",
+        "description": "Gets a deny assignment by ID.",
+        "parameters": [
+          {
+            "name": "denyAssignmentId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The fully qualified deny assignment ID. For example, use the format, /subscriptions/{guid}/providers/Microsoft.Authorization/denyAssignments/{denyAssignmentId} for subscription level deny assignments, or /providers/Microsoft.Authorization/denyAssignments/{denyAssignmentId} for tenant level deny assignments.",
+            "x-ms-skip-url-encoding": true
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns the deny assignment.",
+            "schema": {
+              "$ref": "#/definitions/DenyAssignment"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "GetConfigurations": {
+            "$ref": "./examples/GetDenyAssignmentById.json"
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Authorization/denyAssignments": {
+      "get": {
+        "tags": [
+          "DenyAssignments"
+        ],
+        "operationId": "DenyAssignments_ListForScope",
+        "description": "Gets deny assignments for a scope.",
+        "parameters": [
+          {
+            "name": "scope",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The scope of the deny assignments.",
+            "x-ms-skip-url-encoding": true
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/FilterParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns an array of deny assignments.",
+            "schema": {
+              "$ref": "#/definitions/DenyAssignmentListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "#/definitions/DenyAssignmentFilter",
+        "x-ms-examples": {
+          "GetConfigurations": {
+            "$ref": "./examples/GetDenyAssignmentByScope.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DenyAssignmentFilter": {
+      "properties": {
+        "denyAssignmentName": {
+          "type": "string",
```

Comment:
from the endpoint definitions it seems that all the data returned form the service should be readonly (unmodifiable on the client side). please consider adding `"readOnly": true,` to the properties.

Reply:
Marking a property as "readOnly" is useful when GET returns more properties than used in POST/PUT because then you can use the same schema in both GET and POST/PUT and mark the extra properties as readOnly. But in this case we only have GET requests. So, I don't see any need to mark each property as readOnly.